### PR TITLE
Fix dp_dos for non-collinear cases

### PR DESCRIPTION
**Pull request by aradi**

### CODE REVIEW

File: tools/dptools/src/dptools/bandout.py [hunk 1]
```diff
@@ -62,35 +65,35 @@ def fromfile(cls, fobj):
...
             tmp = np.array(vals.split(), dtype=float)
-
-            # Only keep last two data columns
-            # (optional first column, if present, contains sequential numbering)
             nrows = vals.strip().count('\n') + 1
             ncols = len(tmp) // nrows
-            tmp.shape = (nrows, ncols)
-            tmp = tmp[:, ncols - 2 : ncols]
 
+            # Optional first row might contain sequential numbering.
```
Comment (charlymedrano7) [inline tools/dptools/src/dptools/bandout.py:80]: First row or first column? I am not sure if I understand correctly the following block. For me it makes sense if the first *column* would be sequential numbers. If is that the case and I understood correctly, the script just skip that first column, right? My second question is, when do we have those sequential numbering? Is a matter of previous versions output?

**Review by charlymedrano7 — COMMENTED** [review]
I tested already in some non-collinear systems and seems to work fine. When I artificially created a pdos.out with 7 columns it shows me the proper error.
My only question is in a comment made in the bandout.py that I am not sure if I understood correctly.

File: tools/dptools/src/dptools/bandout.py (hunk 1 above)
Comment (aradi) [reply tools/dptools/src/dptools/bandout.py]: That's a typo, I meant first column of course. Thx, corrected.

As for the sequential numbering: Eigenlevels in `band.out` are numbered for easier interpretation by humans. In the `ProjectedStates` output files, however, this enumeration is missing. Therefore, the script must be able to handle both cases.

Comment (charlymedrano7) [reply tools/dptools/src/dptools/bandout.py]: You are welcome. Understood. Thanks, Bálint!

**Pull request merged**